ui/components/app/alert-system/multiple-alert-modal/multiple-alert-modal.tsx

```diff
@@ -148,8 +148,12 @@ export function MultipleAlertModal({
 }: MultipleAlertModalProps) {
   const { isAlertConfirmed, alerts } = useAlerts(ownerId);
 
+  const initialAlertIndex = alerts.findIndex(
+    (alert: Alert) => alert.key === alertKey,
+  );
+
   const [selectedIndex, setSelectedIndex] = useState(
-    alerts.findIndex((alert: Alert) => alert.key === alertKey),
+    initialAlertIndex === -1 ? 0 : initialAlertIndex,
   );
```

Comment:
May be we move this to hook `useAlerts`. Hook can be passed `selectedAlertKey`.

Reply:
This was simply to make it more readable, plus handle if the alert key isn't found instead of throwing.